hoodie-client/src/main/java/com/uber/hoodie/table/HoodieMergeOnReadTable.java

```diff
@@ -94,4 +114,99 @@ public HoodieMergeOnReadTable(HoodieWriteConfig config,
+                        switch (instant.getAction()) {
+                            case HoodieTimeline.COMMIT_ACTION: case HoodieTimeline.COMPACTION_ACTION:
+                                try {
+                                    Map<FileStatus, Boolean> results = super.delete(partitionPath, Arrays.asList(commit));
+                                    hoodieRollbackStats = HoodieRollbackStat.newBuilder().withPartitionPath(partitionPath)
+                                            .withDeletedFileResults(results).build();
+                                    break;
+                                } catch (IOException io) {
+                                    throw new UncheckedIOException("Failed to rollback for commit " + commit, io);
+                                }
+                            case HoodieTimeline.DELTA_COMMIT_ACTION:
+                                try {
+                                    HoodieCommitMetadata commitMetadata = HoodieCommitMetadata
+                                            .fromBytes(this.getCommitTimeline().getInstantDetails(new HoodieInstant(true, instant.getAction(), instant.getTimestamp())).get());
+
+                                    // read commit file and (either append delete blocks or delete file)
+                                    Map<FileStatus, Boolean> filesToDeletedStatus = new HashMap<>();
+                                    Map<FileStatus, Long> filesToNumBlocksRollback = new HashMap<>();
+
+                                    commitMetadata.getPartitionToWriteStats().get(partitionPath).stream().forEach(wStat -> {
+                                        HoodieLogFormat.Writer writer = null;
+                                        try {
+                                            writer = HoodieLogFormat.newWriterBuilder()
+                                                    .onParentPath(new Path(this.getMetaClient().getBasePath(), partitionPath))
+                                                    .withFileId(wStat.getFileId()).overBaseCommit(wStat.getPrevCommit())
+                                                    .withFs(FSUtils.getFs()).withFileExtension(HoodieLogFile.DELTA_EXTENSION).build();
+
+                                            Long numberOfCommandBlocks = 0L;
+                                            if (writer.getCurrentSize() > 0) {
+                                                numberOfCommandBlocks++;
+                                                if (wStat.getNumDeletes() > 0)
+                                                    numberOfCommandBlocks++;
+                                                Long commandBlocks = numberOfCommandBlocks;
+                                                while (numberOfCommandBlocks > 0) {
+                                                    writer.appendBlock(new HoodieCommandBlock(HoodieCommandBlock.HoodieCommandBlockTypeEnum.ROLLBACK_PREVIOUS_BLOCK));
```

Comment:
I think its best to include the `instantTime` with every log block.. I understand that context is implicit, but does not hurt us much to have this explicit..  It will be much easier to this now than later.. WDYT?

Reply:
Agreed, I think every log block needs to have some metadata associated with it, at the moment it just captures data. `instantTime` can be a good metadata contributor.